src/main/scala/exercise3/ZioMain.scala

```diff
@@ -0,0 +1,82 @@
+package exercise3
+
+import exercise2.lib.StdAudio
+import exercise2.{Queue, QueueLike}
+import zio.{ExitCode, RIO, UIO, URIO, ZIO}
+
+import zio.console._
+import zio.random._
+import zio.stream._
+
+object ZioMain extends zio.App {
+
+  val ENERGY_DECAY_FACTOR = 0.996
+
+  override def run(args: List[String]): URIO[Random with Console, ExitCode] = program.exitCode
+
+  val program: RIO[Random, Unit] = {
+    for {
+      whiteNoise <- whiteNoise()
+      _ <- loop(whiteNoise)
+    } yield ()
+  }
+
+  def play(sample: Double): UIO[Unit] = ZIO.succeed(StdAudio.play(sample))
+
+  /**
+   * Karplus-Strong Part 1.2
+   *
+   * Create a function def whiteNoise(frequency:Int=440, volume:Double = 1.0): Queue[Double]
+   * that returns a queue containing a total of frequency elements of random values between .5 and -.5 multiplied by volume.
+   * Frequency must be greater than zero and volume is between 0 and 1.
+   *
+   * Modify the whiteNoise method, so it uses zio.random for the generation and returns a URIO of Random and Queue[Double].
+   *
+   * mapM: https://zio.dev/docs/datatypes/concurrency/queue#zqueuemapm
+   * Sink: https://zio.dev/docs/datatypes/stream/sink
+   * */
+  def whiteNoise(frequency: Int = 440, volume: Double = 1.0): URIO[Random, Queue[Double]] = {
+    Stream.fromIterable(1 to frequency)
```

Comment:
Interesting solution! `Stream` is deprecated in favor for `LazyList`, which can be created similarly -  ```LazyList.from(1 to frequency)```

Here are some other ways to do it:
using `ZIO.foreach`
```scala
for {
      arr <- ZIO.foreach((1 to frequency).toArray)(_ => nextDoubleBetween(-0.5, 0.5))
      queue = arr.map(x => x * volume).foldLeft(Queue.empty[Double]())((acc, c) => acc.enqueue(c))
    } yield queue
```
using `ZIO.replicateM`
```scala
 for {
      arr <- ZIO.replicateM(frequency)(nextDoubleBetween(-0.5, 0.5))
      queue = arr.map(x => x * volume).foldLeft(Queue.empty[Double]())((acc, c) => acc.enqueue(c))
    } yield queue
```